plugins/OnlineSales.Plugin.AzureAD/AzureADPlugin.cs

```diff
@@ -18,20 +18,30 @@ public void ConfigureServices(IServiceCollection services, IConfiguration config
     {
         var administratorsGroupId = configuration.GetValue<string>("AzureAd:GroupsMapping:Administrators");
 
-        services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
-                    .AddMicrosoftIdentityWebApi(configuration, subscribeToJwtBearerMiddlewareDiagnosticsEvents: true);
+        services.ConfigureAuth(configuration);
 
         services.AddAuthorization(options =>
         {
             options.AddPolicy("Administrators", opts =>
             {
-                opts.RequireClaim(
-                    "http://schemas.microsoft.com/ws/2008/06/identity/claims/role",
-                    administratorsGroupId ?? Guid.NewGuid().ToString());
+                if (!string.IsNullOrEmpty(administratorsGroupId))
+                {
+                    opts.RequireClaim(
+                        "http://schemas.microsoft.com/ws/2008/06/identity/claims/role",
+                        administratorsGroupId);
+                    return;
+                }
+
+                throw new MissingAdminGroupException("AzureAd:GroupsMapping:Administrators field is required.");
             });
         });
+        services.Configure<CookiePolicyOptions>(options =>
+        {
+            options.MinimumSameSitePolicy = SameSiteMode.None;
+            options.Secure = CookieSecurePolicy.Always;
+        });
     }
-    
+
     public void ConfigureSwagger(SwaggerGenOptions options, OpenApiInfo settings)
     {
         options.AddSecurityDefinition("Bearer", new OpenApiSecurityScheme()

```

Comment:
@Exmirai new cookie based authentication should also be supported in swagger UI. Add it here.